src/app/inventory/store/sockets.ts

```diff
@@ -535,10 +535,17 @@ function buildCachedDefinedPlug(defs: D2ManifestDefinitions, plugHash: number):
   const cachedValue = definedPlugCache[plugHash];
   // The result of buildDefinedPlug can be null, we still consider that a cached value.
   if (cachedValue !== undefined) {
-    return cachedValue;
+    // We mutate cannotCurrentlyRoll and attach stats in this module so we need to spread the object
+    // We also run DimItems through immer in the store, which means these get frozen. This essentially
+    // unfreezes it in that situation. It only seems to be an issue for fake items in loadouts.
+    // TODO (ryan) lets fine a way around this
+    return cachedValue ? { ...cachedValue } : cachedValue;
```

Comment:
```suggestion
    return cachedValue ? { ...cachedValue } : null;
```
is this correct? i think it would be nice for more explicitly demonstrating the "intention"

Reply:
yeah it is, will commit

Reply:
This is what fixes the issue with the error spam about frozen stats and `cannotCurrentlyRoll` being modified.